src/main/java/com/google/sps/data/User.java

```diff
@@ -0,0 +1,26 @@
+package com.google.sps.meltingpot.data;
+
+import com.google.api.core.ApiFuture;
+import com.google.cloud.firestore.DocumentReference;
+import com.google.cloud.firestore.DocumentSnapshot;
+import com.google.sps.meltingpot.data.DBUtils;
+
+public class User {
+  public static final String CREATED_RECIPES_KEY = "created-recipe-ids";
+
+  public static boolean createdRecipe(String userId, String recipeId) {
+    DocumentReference userRef = DBUtils.user(userId);
+    ApiFuture<DocumentSnapshot> userFuture = userRef.get();
+    DocumentSnapshot user = DBUtils.blockOnFuture(userFuture);
+    if (!user.exists()) {
+      return false;
+    }
+    Boolean userCreatedRecipe =
+        user.getBoolean(DBUtils.getNestedPropertyName(CREATED_RECIPES_KEY, recipeId));
+    // note that userCreatedRecipe is a Boolean, not a boolean
+    if (userCreatedRecipe == null || userCreatedRecipe == false) {
+      return false;
+    }
+    return true;
```

Comment:
I might argue this would be more readable as
return userCreatedRecipe != null && userCreatedRecipe == true;

but I wonder if that's really necessary, does user.getBoolean ever return a null? It'd be nice if it didn't...

Reply:
I believe it throws null if the field isn't found, so we do have to take that into account.
Done in 13fde54.